### ISSUE TITLE
Kitty-0.30.1/lib/kitty/terminfo: command not found

### ISSUE THREAD

**Issue by Schweber** [NixOS/nixpkgs#260427]
### Describe the bug
Since upgrading from 0.28.1 to 0.30.1 i get this error when invoking sudo. With 0.28.1 and other terminal emulators everthing works as expected.

```
~  sudo -i                                                                                                                                      
doas (sperber@nixos) password:
doas: TERMINFO=/nix/store/444jzask4nm2i9vgih417mnfwlzzvd9i-kitty-0.30.1/lib/kitty/terminfo: command not found
```

### Notify maintainers

@tex @rvolosatovs @Luflosi @adamcstephens 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

This yields no output. I am on NixOS 23.05.

**Comment by adamcstephens** [comment]
Are you aliasing sudo to doas? does doesn't even appear to support `-i` so I'm confused by the output. Can you try with just regular sudo or doas?

```
─❯ doas sh
sh-5.2# echo $TERM
xterm-kitty
sh-5.2# exit

─❯ sudo -i
[sudo] password for adam: 
[root@blank:~]# echo $TERM
xterm-kitty
```

**Comment by Schweber** [comment]
Yes, i alias sudo to doas with `(pkgs.writeScriptBin "sudo" ''exec doas "$@"'')` under `  environment.systemPackages = with pkgs; [`. 

You were right to suspect this to be relevant, because running a command in kitty 0.30.1 with doas works, while running it with "sudo" doesn't:  

```
 ~  sudo nixos-rebuild switch --upgrade-all                        
doas (sperber@nixos) password:
doas: TERMINFO=/nix/store/444jzask4nm2i9vgih417mnfwlzzvd9i-kitty-0.30.1/lib/kitty/terminfo: command not found

 ~  doas nixos-rebuild switch --upgrade-all                  
unpacking channels...
unpacking channels...
unpacking channels...
building Nix...
building the system configuration...
^Cerror: interrupted by the user
Interrupt
```

However, with the previous versions of `kitty` and with `foot` this error was/is not generated. Here is the output of `foot`:

```
 ~  sudo nixos-rebuild switch --upgrade-all                                                                                                       ✔ 
doas (sperber@nixos) password: 
unpacking channels...
unpacking channels...
unpacking channels...
building Nix...
building the system configuration...
^Cerror: interrupted by the user
Interrupt
```

Was maybe something changed in the package, because there is now the error about the missing `terminfo`?

I understand that this is an edge case and may be irrelevant but i switched to wayland today and kitty is giving me trouble by not recognizing the keys it did under X11 so i'm trying to rule out other reasons for that.

**Comment by adamcstephens** [comment]
I suspect you have other modifications in your environment, since you are getting `command not found` not an error about TERM. Furthermore, I reviewed the kitty changes since 0.28.1 and see nothing related to the terminfo files.

The only reproducer I can find is if I pass TERMINFO to doas. I'm not sure if doas actually supports this syntax, but sudo definitely does.

```
─❯ doas TERMINFO=/nix/store/444jzask4nm2i9vgih417mnfwlzzvd9i-kitty-0.30.1/lib/kitty/terminfo ls
doas: TERMINFO=/nix/store/444jzask4nm2i9vgih417mnfwlzzvd9i-kitty-0.30.1/lib/kitty/terminfo: command not found
```

If kitty keyboard shortcuts aren't working after moving from xorg to wayland, I'd suspect the wayland compositor is intercepting them. I use kitty on both without issue.

**Comment by Schweber** [comment]
Alright, thank you for looking it! In this case i will change my aliases from sudo to doas and try not to think about it anymore. ;-)

> If kitty keyboard shortcuts aren't working after moving from xorg to wayland, I'd suspect the wayland compositor is intercepting them. I use kitty on both without issue.

I just found the error, it was my mistake. I had `kitty -1 --class kitty_hx sh -c hx` as a shortcut in X11 and transposed it to `kitty -1 --class kitty_hx -c sh hx` in wayland and didn't realize why it showed strange behaviour when using this shortcut. I now got rid of the `-c` altogether. :-)
